### PR TITLE
Refactor DynamoDB create-table code

### LINKED ISSUE #6927: Use AWS SDK DynamoDB waiters when creating KV store
- Always call CreateTable and just accept "already exists" as the table existing
- Use a waiter configured for near-constant time (instead of rolling our own)
- Log how long creating the table took

### PR DESCRIPTION
**Pull request by arielshaqed**
- Don't check if the table exists, just try to create it.  We anyway need to accept that `CreateTable` may fail because the table exists; in an average cluster setup this will happen on n-1 of the members!
- Use an AWS-provided waiter to wait for the table existing rather than writing our own.
- Report how long it took.

Fixes #6927.

### CODE REVIEW

**Comment by arielshaqed** [comment]
@johnnyaug  since you're working on similar code elsewhere, we should probably try to steal code between us.

**Review by nopcoder — APPROVED** [review]
LGTM - minor comment about the duration field log format

File: pkg/kv/dynamodb/store.go [hunk 1]
```diff
@@ -148,12 +148,13 @@ func isTableExist(ctx context.Context, svc *dynamodb.Client, table string) (bool
...
 func setupKeyValueDatabase(ctx context.Context, svc *dynamodb.Client, params *kvparams.DynamoDB) error {
-	// main kv table
-	exist, err := isTableExist(ctx, svc, params.TableName)
-	if exist || err != nil {
-		return err
-	}
+	log := logging.FromContext(ctx).WithField("table_name", params.TableName)
+	start := time.Now()
+	defer func() {
+		log.WithField("took", fmt.Sprint(time.Since(start))).Info("Setup time")
```
[the remark this answers is not in the document]
Comment (arielshaqed) [reply pkg/kv/dynamodb/store.go:153]: We consistently see numbers in logrus JSON outputs for our existing "took" fields; if this one works out I'll patch the others too.

This one is from a sample Esti run as seen on lakeFS Cloud (staging):
```json
{
  "client": "ayti",
  "file": "usr/local/go/src/net/http/server.go:2122",
  "func": "net/http.HandlerFunc.ServeHTTP",
  "host": "ayti-ci-2495-208c69--esti.us-east-1.lakefscloud.ninja",
  "level": "info",
  "log_audit": true,
  "method": "GET",
  "msg": "HTTP call ended",
  "operation_id": "GetRepository",
  "path": "/api/v1/repositories/ayti-test-repo",
  "request_id": "cc20bafb-5299-419f-97e0-d0899eeb145a",
  "sent_bytes": 30,
  "service_name": "rest_api",
  "source_ip": "10.100.32.170:57848",
  "status_code": 500,
  "time": "2023-10-28T11:49:07Z",
  "took": 3035101,
  "user": "esti"
}

```

[the remark this answers is not in the document]
Comment (arielshaqed) [reply pkg/kv/dynamodb/store.go:153]: I think that it's because `time.Duration` is an integer type, which probably confuses Go and/or logrus. :shrug.

**Comment by arielshaqed** [comment]
Thanks!

Pulling...

**Pull request merged**